test/suite/run.py

```diff
@@ -511,12 +529,31 @@ def error(exitval, prefix, msg):
     if debug:
         import pdb
         pdb.set_trace()
+    if batchtotal != 0:
+        # For test batching, we want to split up all the tests evenly, and
+        # spread out the tests, so each batch contains tests of all kinds. We'd
+        # like to prioritize the lowest scenario numbers first, so if there's a
+        # failure, we won't have to do all X thousand of some test's scenarios
+        # before we see a failure in the next test.  To that end, we define a
+        # sort function that sorts by scenario first, and test name second.
+        hugetests = set()
+        def batchsort(test):
+            s = 0
+            name = test.simpleName()
+            if hasattr(test, 'scenario_number'):
+                s = test.scenario_number
+                if s > 1000:
+                    hugetests.add(name)    # warn for too many scenarios
+            return (s, test.simpleName())  # sort by scenerio number first
+        all_tests = sorted(tests, key = batchsort)
+        for name in hugetests:
+            print("WARNING: huge test " + name + " has > 1000 scenarios")
```

Comment:
Does this warning imply the test name has too many scenarios, and should be split?

Reply:
Yes, exactly what this warning implies.  There's currently one test (compat02??) that goes beyond 1000.  I think once we "prune" that, we should make this warning into an error, but only if the "-long" option is not used.  I'll add the "-long" option into this check, so it doesn't appear for long runs.